src/test/groovy/com/palantir/gradle/shadowjar/ShadowJarPluginIntegrationSpec.groovy

```diff
@@ -150,6 +150,38 @@ class ShadowJarPluginIntegrationSpec extends IntegrationSpec {
         assert !jarEntryNames.contains(relocatedClass('org/slf4j/impl/Log4jLoggerFactory.class'))
     }
 
+    def 'should support multi-release jars (MRJARS, https://www.baeldung.com/java-multi-release-jar)'() {
+        when:
+        buildFile << """
+            repositories {
+                mavenCentral()
+            }
+            
+            dependencies {
+                shadeTransitively 'one.util:streamex:0.7.3'
+            }
+            
+            task extractForAssertions(type: Copy) {
+                dependsOn publishNebulaPublicationToTestRepoRepository
+                from zipTree("${MAVEN_ROOT}/com/palantir/bar-baz_quux/asd-fgh/2/asd-fgh-2.jar")
+                into "\$buildDir/extractForAssertions"
+            }
+        """.stripIndent()
+
+        then:
+        writeHelloWorld()
+        runTasksAndCheckSuccess('extractForAssertions')
+
+        def jarEntryNames = shadowJarFile().stream()
+                .map({it.name})
+                .collect(Collectors.toCollection({new LinkedHashSet()}))
+
+        assert jarEntryNames.contains(
+                'META-INF/versions/9/shadow/com/palantir/bar_baz_quux/asd_fgh/one/util/streamex/VerSpec.class')
+        assert jarEntryNames.contains(
+                'META-INF/versions/9/shadow/com/palantir/bar_baz_quux/asd_fgh/one/util/streamex/Java9Specific.class')
```

Comment:
I added additional asserts in #70 which illustrate another bug: we're not inheriting the `Multi-Release: true` manifest line, so the versioned components won't be loaded.

Reply:
Ok think this is fixed now :)